src/main/java/roomescape/service/ReservationService.java

```diff
@@ -63,25 +67,58 @@ public List<ReservationResponse> getFilteredReservations(final Long memberId, fi
                 .toList();
     }
 
-    private void validateDuplicateReservation(final LocalDate localDate, final long timeId, final long themeId) {
-        if (reservationRepository.existByDateAndTimeIdAndThemeId(localDate, timeId, themeId)) {
-            throw new IllegalArgumentException("[ERROR] 이미 존재하는 예약 입니다.");
-        }
-    }
-
-    public List<MyPageReservationResponse> getReservationsByMemberId(Long memberId) {
+    @Transactional
+    public List<MyPageReservationResponse> getReservationsByMemberId(final Long memberId) {
         final Member member = memberRepository.findById(memberId)
                 .orElseThrow(() -> new NoSuchElementException("[ERROR] 존재하지 않는 사용자 입니다."));
-        List<Reservation> myReservations = reservationRepository.findByMemberId(member.getId());
-        return myReservations.stream()
+        final List<Reservation> myReservations = reservationRepository.findByMemberId(member.getId());
+        final List<ReservationWaiting> myReservationWaitings = reservationWaitingRepository.findByMemberId(
+                member.getId());
+        final List<MyPageReservationResponse> myPageReservationResponses = myReservations.stream()
                 .map(MyPageReservationResponse::from)
+                .collect(Collectors.toList());
+        List<MyPageReservationResponse> myPageReservationWaitingResponses = myReservationWaitings.stream()
+                .map(myReservationWaiting -> MyPageReservationResponse.of(myReservationWaiting,
+                        getWaitingOrderByMember(myReservationWaiting.getMember())))
                 .toList();
+        myPageReservationResponses.addAll(myPageReservationWaitingResponses);
+        return myPageReservationResponses;
```

Comment:
```suggestion
        return toResponse(myReservations, myReservationWaitings);
```
사소하지만, 이렇게 reponse 반환 로직이 조금 복잡한 경우 저는 메서드를 분리하면 
핵심로직과 response 조합 로직이 분리되어 가독성이 조금 좋아질 것 같아요!

Reply:
오 굿이네요 좋습니다!